### PR TITLE
Remove example solution code

### PR DESCRIPTION
**Pull request by alexandre-allard**
**Component**: examples, eve

**Context**:
Since now example-solution has its own repository https://github.com/scality/metalk8s-solution-example, we can clean up everything related to it from the MetalK8s repository.

**Summary**:
* Remove example-solution code
* Remove example-solution build in CI
* Adapt solution post-merge in CI

**Acceptance criteria**: 
Solution post-merge tests must still work.

### CODE REVIEW

File: eve/main.yml [hunk 1]
```diff
@@ -716,11 +717,14 @@ stages:
...
-            - build-example-solution
-          haltOnFailure: true
+      - SetProperty:
+          name: Set Example Solution version property
+          property: example_solution_version
+          value: 1.0.0
+      - SetProperty:
+          name: Set Example Solution version next property
+          property: example_solution_version_next
+          value: 1.0.1
```
Comment (TeddyAndrieux) [inline eve/main.yml:833]: This one does not exists yet right ?

Comment (alexandre-allard) [reply eve/main.yml:833]: No, I need to release it.

Comment (alexandre-allard) [reply eve/main.yml:833]: Done

File: eve/main.yml [hunk 2]
```diff
@@ -2460,7 +2389,9 @@ stages:
             FILE_SOURCE: >-
               example-solution-%(prop:example_solution_version)s.iso
             FILE_DEST: example-solution.iso
-            BASE_URL: "%(prop:artifacts_private_url)s"
+            BASE_URL: "http://artifacts/builds/github: \
+                       scality:metalk8s-solution-example: \
+                       promoted-%(prop:example_solution_version)s"
```
Comment (gdemonet) [inline eve/main.yml]: Won't this resolve to `http://artifacts/builds/github: scality:metalk8s-solution-example: promoted-%(prop:example_solution_version)s` (with spaces), and cause problems down the line?

```suggestion
            BASE_URL: "http://artifacts/builds/github:\
                       scality:metalk8s-solution-example:\
                       promoted-%(prop:example_solution_version)s"
```

Comment (alexandre-allard) [reply eve/main.yml]: Yes, totally.

**Review by gdemonet — DISMISSED** [review]

**Comment by NicolasT** [comment]
@bert-e status

File: ui/cypress/integration/e2e/Solutions/steps.js [hunk 3]
```diff
@@ -16,7 +16,7 @@ When(
       '/api/kubernetes/api/v1/namespaces?labelSelector=solutions.metalk8s.scality.com/environment',
     ).as('getEnvironmentsList');
 
-    cy.get('.sc-sidebar-item').eq(2).click(); // go to solutions list
+    cy.get('.sc-sidebar-item').eq(3).click(); // go to solutions list
```
Comment (gdemonet) [inline ui/cypress/integration/e2e/Solutions/steps.js]: That's why we should provide `data-cy` attributes on all items (cc @ChengYanJin :stuck_out_tongue_winking_eye:)

File: eve/main.yml [hunk 4]
```diff
@@ -2467,6 +2467,7 @@ stages:
     worker:
       type: local
     steps:
+      - ShellCommand: *add_final_status_artifact_failed
```
Comment (gdemonet) [inline eve/main.yml]: Is this related to this PR (I see it appears in the `generate-snapshot` stage)? Wasn't this part of #3084 already?

Comment (alexandre-allard) [reply eve/main.yml]: Hmm weird, I'm not sure I did put this in this stage... Maybe due to a rebase ? (or I just made a mistake).
Anyway, it's definitely not related to this PR and as you commented, it has already been fixed, I'll remove this line.

Comment (alexandre-allard) [reply eve/main.yml]: done

**Review by gdemonet — DISMISSED** [review]

File: ui/cypress/integration/e2e/Solutions/steps.js (hunk 3 above)
Comment (gdemonet) [inline ui/cypress/integration/e2e/Solutions/steps.js]:
```suggestion
    cy.get('[data-cy="sidebar_item_environments"]').click(); // go to solutions list
```

**Review by gdemonet — APPROVED** [review]

**Pull request merged**